### ISSUE TITLE
The program cannot stop after 1000 epoch

### ISSUE THREAD

**Issue by Bigsealion** [MIC-DKFZ/nnUNet#842]
Hi,
Now, I have finished the training of a 2d nnUNet (all 1000 epoch), but the program does not stop and the GPU memory is still occupied. (In fact, the program is no longer running, and `ps -aux` show the STAT is `S` or `Sl`) 
Therefore, I can only use the `kill` command to forcibly stop the process.

My training command is:
`nohup nnUNet_train 2d nnUNetTrainerV2 Task604_Seg 3 --npz &> ./task604_2d_f3.txt &`
Did command `nohup` or `&` cause this error, or else?

In the default log file (training_log_2021_11_4_15_55_53.txt), the last paragraph is as follows:
`
2021-11-05 13:25:22.656965: 
epoch:  999 
2021-11-05 13:26:33.520940: train loss : -0.9004 
2021-11-05 13:26:39.819033: validation loss: -0.8059 
2021-11-05 13:26:39.820221: Average global foreground Dice: [0.9105, 0.7286] 
2021-11-05 13:26:39.820441: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-11-05 13:26:40.250808: lr: 0.0 
2021-11-05 13:26:40.251173: saving scheduled checkpoint file... 
2021-11-05 13:26:40.286398: saving checkpoint... 
2021-11-05 13:26:40.507148: done, saving took 0.26 seconds 
2021-11-05 13:26:40.525447: done 
2021-11-05 13:26:40.525670: This epoch took 77.868647 s
 
2021-11-05 13:26:40.556237: saving checkpoint... 
2021-11-05 13:26:40.960068: done, saving took 0.43 seconds 
2021-11-05 13:31:52.777114: finished prediction 
2021-11-05 13:31:52.777730: evaluation of raw predictions 
`

In my log file, the last paragraph is as follows:
`
SegData_743 (2, 91, 109, 91)
debug: mirroring True mirror_axes (0, 1)
2021-11-05 13:31:52.777114: finished prediction
2021-11-05 13:31:52.777730: evaluation of raw predictions
`

Thanks!

**Comment by FabianIsensee** [comment]
Hi, this should not happen. It should just exit once it is done. Maybe it was still evaluating the predictions? So you have a summary.json file in the output folder?

**Comment by Bigsealion** [comment]
No, there is no `summary.json` in output folders `validation_raw`
But training has long over (the last generated file in `validation_raw` is 13:31, but kill time is after 16:00)

**Comment by FabianIsensee** [comment]
can you please rerun validation (use the `-val` flag) and see if it now finishes? Please monitor RAM usage while this is running. Is the RAM full at some point?

**Comment by dojoh** [comment]
Hello,

Did you manage to resolve the problem or do you need further assistance?
Otherwise, I will close the issue.

Cheers
Ole

**Comment by Meruja** [comment]
Hi

I encountered the same issue; Here is the output.
2024-02-19 00:28:18.851366: finished prediction
2024-02-19 00:28:18.868930: evaluation of raw predictions
2024-02-19 00:31:04.906207: determining postprocessing 

It's been 9hrs now, and I don't see any progress in postprocessing. And there's no summary.json file in the output folder.

Thanks

**Comment by FabianIsensee** [comment]
Is this on a public dataset? It would be useful to be able to reproduce this

**Comment by Meruja** [comment]
> Is this on a public dataset? It would be useful to be able to reproduce this

No, it's a private data. CBCT scans.
For the rest of the folds, I didn't receive any issues. I encountered a similar issue when I tried to do the plan and process but fixed it with -tf 1 -tl 1.

**Comment by FabianIsensee** [comment]
Sounds like a RAM issue to me. Can you try upgrading to the latest version of nnU-Net? It automatically notices when workers disappear because of that

**Comment by Meruja** [comment]
I was able to fix it. It's a RAM issue.
